net/tcp/tcp_send_buffered.c

```diff
@@ -484,6 +485,26 @@ static uint16_t psock_send_eventhandler(FAR struct net_driver_s *dev,
                         wrb, TCP_WBSEQNO(wrb), TCP_WBPKTLEN(wrb));
                 }
             }
+          else if (ackno == TCP_WBSEQNO(wrb))
+            {
+              /* Duplicate ACK? Retransmit data if need */
+
+              if (++TCP_WBNACK(wrb) ==
+                  CONFIG_NET_TCP_FAST_RETRANSMIT_WATERMARK)
+                {
+                  /* Do fast retransmit */
+
+                  rexmit = true;
+                }
+              else if ((TCP_WBNACK(wrb) >
+                       CONFIG_NET_TCP_FAST_RETRANSMIT_WATERMARK) &&
+                       TCP_WBNACK(wrb) == sq_count(&conn->unacked_q) - 1)
```

Comment:
@anchao can you explain the intention of this condition?

Reply:
In some cases, retransmission packets may be lost again. Here we reset the retransmission count and prepare to resend

Reply:
what does "TCP_WBNACK(wrb) == sq_count(&conn->unacked_q) - 1" mean?

Reply:
If the received dupack count is equal to unacked_q. That means that the ack of all packets after this packet loss has been received, so the counting of DUP ack will be restarted.

Reply:
in that case, we have likely already started fast retransmit.
* do you mean the segment fast-retransmitted has lost again? such a case is simpler to leave to the usual "slow retransmit".
* this implementation of fast retransmit can retransmit multiple segments. (usually fast retransmit only retransmit one segment. but to me this implementation doesn't seem to have such a mechanism.) so the calculation here doesn't seem to make much sense to me.